doc/internal/man3/OPTIONS.pod

```diff
@@ -0,0 +1,269 @@
+ int opt_ulong(const char *value, unsigned long *result);
+
+ int opt_isdir(const char *name);
+
+ int opt_format(const char *s, unsigned long flags, int *result);
+ int opt_string(const char *name, const char **options);
+ int opt_pair(const char *name, const OPT_PAIR* pairs, int *result);
+
+ int opt_num_rest(void);
+ char **opt_rest(void);
+
+=head1 DESCRIPTION
+
+The functions on this page provide a common set of option-parsing for
+the OpenSSL command and the internal test programs.
+It is intended to be used like the standard getopt(3) routine, except
+that multi-character flag names are supported, and a variety of parsing
+and other utility functions are also provided.
+
+Programs that use this should make sure to set the appropriate C<-I>
+flag.
+
+These routines expect a global B<BIO> named B<bio_err> to point to
+the equivalent of B<stderr>. This is already done in the OpenSSL
+application.
+
+=head2 Data Types
+
+Each program should define, near the main() routine, an enumeration
+that is the set of options the program accepts. For example:
+
+    typedef enum OPTION_choice {
+        OPT_ERR = -1, OPT_EOF = 0, OPT_HELP,
+        OPT_YES, OPT_NAME, OPT_COUNT, OPT_OFILE
+    } OPTION_CHOICE;
+
+The first two lines must appear exactly as shown. The third line is a sample
+set of flags, and the closing C<typedef> name is used for error-checking
+as discussed below.
+
+The B<OPTIONS> C<typedef> specifies an option: what type of argument
+it takes (if any), and an optional "help" string.  It is a C<struct>
+containing these fields:
+
+    const char *name;
+    int retval;
+    int valtype;
+    const char *helpstr;
+
+The B<name> is the name of the option that the user would type. Options
+are words preceeded with a minus sign. If the user uses two minus signs,
+this is also accepted for compatibility with other GNU software. Some
+names are special, and are described below.
+
+The B<retval> is the value to return if the option is found. It should be
+one of the choices in the enumeration above.
+
+The B<valtype> defines what the option's parameter must be. It should
+be chosen from the following set:
+
+    \0  No value
+    '-' No value
+    's' A text string
+    '/' A directory
+    '<' Name of file to open for input
+    '>' Name of file to open for output
+    'n' A signed number that fits in the C<int> type
+    'p' A nonnegative number that fits in the C<int> type
+    'M' A signed number that fits in the C<intmax_t> type
+    'U' An unsigned number that fits in the C<uintmax_t> type
+    'l' A signed number that fits in the C<long> type
+    'u' An unsigned number that fits in the C<unsigned long> type
+    'c' File of fomer PEM, DER, or S/MIME
+    'F' A file in PEM or DER format
+    'E' Like 'F' but also allows ENGINE
+    'f' Any file format
+
+The B<helpstr> is what to display when the user uses the help flag,
+which should be C<"help">.
+
+A program should declare its options right after the enumeration:
```

Comment:
Please mention that the order of the entries should be the same as in the enumeration, which helps readability and maintainability.